src/lib/Content/Content.php

```diff
@@ -0,0 +1,193 @@
+use EzSystems\EzRecommendationClient\Field\Value;
+use Symfony\Component\Console\Helper\ProgressBar;
+use Symfony\Component\Console\Output\NullOutput;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\HttpFoundation\ParameterBag;
+use Symfony\Component\Routing\RouterInterface;
+
+class Content
+{
+    /** @var \eZ\Publish\API\Repository\ContentService */
+    protected $contentService;
+
+    /** @var \eZ\Publish\API\Repository\ContentTypeService */
+    protected $contentTypeService;
+
+    /** @var \eZ\Publish\API\Repository\LocationService */
+    protected $locationService;
+
+    /** @var \Symfony\Component\Routing\Generator\UrlGeneratorInterface */
+    protected $generator;
+
+    /** @var \EzSystems\EzRecommendationClient\Field\Value */
+    protected $value;
+
+    /** @var int $defaultAuthorId */
+    private $defaultAuthorId;
+
+    /**
+     * @param \eZ\Publish\API\Repository\ContentService $contentService
+     * @param \eZ\Publish\API\Repository\ContentTypeService $contentTypeService
+     * @param \eZ\Publish\API\Repository\LocationService $locationService
+     * @param \Symfony\Component\Routing\RouterInterface $routingGenerator
+     * @param \EzSystems\EzRecommendationClient\Field\Value $value
+     * @param int $defaultAuthorId
+     */
+    public function __construct(
+        ContentServiceInterface $contentService,
+        ContentTypeServiceInterface $contentTypeService,
+        LocationServiceInterface $locationService,
+        RouterInterface $routingGenerator,
+        Value $value,
+        int $defaultAuthorId
+    ) {
+        $this->contentService = $contentService;
+        $this->contentTypeService = $contentTypeService;
+        $this->locationService = $locationService;
+        $this->generator = $routingGenerator;
+        $this->value = $value;
+        $this->defaultAuthorId = $defaultAuthorId;
+    }
+
+    /**
+     * Prepare content array.
+     *
+     * @param array $data
+     * @param \Symfony\Component\HttpFoundation\ParameterBag $options
+     * @param \Symfony\Component\Console\Output\OutputInterface|null $output
+     *
+     * @return array
+     *
+     * @throws \eZ\Publish\API\Repository\Exceptions\NotFoundException
+     * @throws \eZ\Publish\API\Repository\Exceptions\UnauthorizedException
+     */
+    public function prepareContent(array $data, ParameterBag $options, ?OutputInterface $output = null): array
+    {
+        if ($output === null) {
+            $output = new NullOutput();
+        }
+
+        $content = [];
+
+        foreach ($data as $contentTypeId => $items) {
+            $progress = new ProgressBar($output, count($items));
+            $progress->start();
+
+            foreach ($items as $contentValue) {
+                $contentValue = $contentValue->valueObject;
+                $contentType = $this->contentTypeService->loadContentType($contentValue->contentInfo->contentTypeId);
```

Comment:
Should `\eZ\Publish\API\Repository\Exceptions\NotFoundException` or `\eZ\Publish\API\Repository\Exceptions\UnauthorizedException` be handled somehow or is your intention to let it be propagated to the top (Controller/Command)? The same question applies to`ContentTypeController`.